developer-docs-site/docs/guides/sign-a-transaction.md

```diff
@@ -0,0 +1,305 @@
+
+### Signed transaction
+
+A signed transaction consists of:
+
+* A raw transaction, and
+* An authenticator. The authenticator contains a client's public key and the signature of the raw transaction.
+
+This signed transaction is further BCS-serialized and hex-coded (not shown in the diagram in [Overview](#overview) section), after which the signed transaction is ready for submission to the [Aptos REST interface](https://fullnode.devnet.aptoslabs.com/spec.html).
+
+### Multisignature transactions
+
+The Aptos Blockchain supports several signing methods for transactions, including the single signature, the K-of-N multisig, and more.
+
+A K-of-N multisig transaction means that for such a transaction to be executed, at least K out of the N authorized signers have signed the transaction and passed the check conducted by the chain.
+
+Transaction signatures are wrapped in `Authenticator`. The Aptos Blockchain validates the transactions submitted by clients by using the Authenticator data.  See a few examples below:
+
+In Typescript, this is how a single signer authenticator looks like:
+```typescript
+interface Authenticator {
+  public_key: Uint8Array,
+  signature: Uint8Array
+}
+```
+
+A multisig authenticator example is shown below:
+```typescript
+interface  MultiEd25519PublicKey {
+  public_keys: Uint8Array[], // A list of public keys
+  threshold: Uint8, // At least `threshold` signatures must be valid
+}
+
+interface MultiEd25519Signature {
+    signatures: Uint8Array[], // A list of signatures
+    // 4 bytes, at most 32 signatures are supported.
+    // If Nth bit value is `1`, the Nth signature should be provided in `signatures`. Bits are read from left to right
+    bitmap: Uint8Array,
+}
+
+interface MultisigAuthenticator {
+  public_key: MultiEd25519PublicKey,
+  signature: MultiEd25519Signature
+}
+```
+
+## Detailed steps
+
+1. Creating a RawTransaction.
+2. Preparing the message to be signed and signing it.
+3. Creating an Authenticator and a SignedTransaction.
+4. Serializing SignedTransaction.
+
+### Step 1. Creating a RawTransaction
+
+The below example assumes the transaction has a script function payload.
+
+```typescript
+
+interface ModuleId {
+  address: string,
+  name: string
+}
+
+interface ScriptFunction {
+  module: ModuleId,
+  function: string,
+  ty_args: string[],
+  args: Uint8Array
+}
+
+interface RawTransaction {
+  sender: string,
+  sequence_number: number,
+  payload: ScriptFunction,
+  max_gas_amount: number,
+  gas_unit_price: number,
+  expiration_timestamp_secs: number,
+  chain_id: number,
+}
+
+function createRawTransaction(): RawTransaction {
+  const payload: ScriptFunction = {
+    module: {
+      address: "00000000000000000000000000000001",
+      name: "TestCoin"
+    },
+    function: "transfer",
+    ty_args: [],
+    args: [
+      BCS.serialize_str("00000000000000000000000000000002"), // receipient of the transfer
```

Comment:
This is an address, not a string. Has this been tested? An address should be 32-byte fixed length value in bcs afaiu.